ckanext/extractor/interfaces.py

```diff
@@ -81,3 +81,21 @@ def extractor_after_index(self, resource_dict, metadata_dict):
         according to ``ckanext.extractor.indexed_fields``.
         '''
 
+
+class IExtractorRequest(plugins.Interface):
+    '''
+    Plugin into the requests made to download content for extraction.
+
+    The hooks provide a possibility to change the parameters of the request,
+    e.g. if the remote server requires certain headers or authorization tokens
+    or to completly alter the URL.
+
+    '''
+    def extractor_before_request(self, request):
+        '''
+        Change the PreparedRequest object according to your needs
+
+        See the requests documentation for details:
+        http://docs.python-requests.org/en/master/user/advanced/
+        '''
+        return request
```

Comment:
The docs should mention that you need to return a request (which could be the original one after modifying it).